catchup/classBasedPeerSelector.go

```diff
@@ -58,6 +59,9 @@ func (c *classBasedPeerSelector) rankPeer(psp *peerSelectorPeer, rank int) (int,
 		// Peer was in this class, if there was any kind of download issue, we increment the failure count
 		if rank >= peerRankNoBlockForRound {
 			wp.downloadFailures++
+		} else if wp.downloadFailures > 0 {
+			// class usually multiple peers and we do not want to punish the entire class for one peer's failure
+			wp.downloadFailures--
```

Comment:
Basically, somebody did something right so lets not penalize as heavily...I do wonder if it should be 1 for 1 or more like 1 failure vs a few successes to revert...but that adds complexity too.

Reply:
How about:
```
success := rank < peerRankNoBlockForRound
if success {
   wp.downloadFailures = max(wp.downloadFailures-1, 0)
} else {
   wp.downloadFailures++
}
```
I think that's a clearer way of showing downloadFailures goes up for failures, down for success.

Reply:
> I do wonder if it should be 1 for 1 or more like 1 failure vs a few successes to revert...but that adds complexity too.

we have a similar logic in history tracking in `peerSelector`, so I guess it is fine
```golang
		if hs.downloadFailures > 0 {
			hs.downloadFailures--
		}
```

> I think that's a clearer way of showing downloadFailures goes up for failures, down for success.

thanks, applied a slight variation with `failure := rank >= peerRankNoBlockForRound`